src/sentry/message_filters.py

```diff
@@ -0,0 +1,559 @@
+    """
+    Returns the filter state
+
+    IMPORTANT: this function accesses the database, it should NEVER be used by the ingestion pipe.
+    This api is used by the ProjectFilterDetails and ProjectFilters endpoints
+    :param filter_id: the filter Id
+    :param project: the project for which we want the filter state
+    :return: True if the filter is enabled False otherwise
+    :raises: ValueError if filter id not registered
+    """
+    flt = _filter_from_filter_id(filter_id)
+    if flt is None:
+        raise FilterNotRegistered(filter_id)
+
+    filter_state = ProjectOption.objects.get_value(project=project, key=u'filters:{}'.format(flt.spec.id))
+
+    if filter_state is None:
+        return flt.spec.default_enabled
+
+    if flt == _legacy_browsers_filter:
+        # special handling for legacy browser state
+        if filter_state == '1':
+            return True
+        if filter_state == '0':
+            return False
+        return filter_state
+    else:
+        return filter_state == "1"
+
+
+class FilterNotRegistered(Exception):
+    pass
+
+
+# ########################################################################################################
+# ########################################### Implementation #############################################
+# ########################################################################################################
+
+
+def _filter_from_filter_id(filter_id):
+    """
+    Returns the corresponding filter for a filter id or None if no filter with the given id found
+    """
+    for flt in get_all_filters():
+        if flt.spec.id == filter_id:
+            return flt
+    return None
+
+
+class _FilterSerializer(serializers.Serializer):
+    active = serializers.BooleanField()
+
+
+class _FilterSpec(object):
+    """
+    Data associated with a filter, it defines its name, id, default enable state and how its  state is serialized
+    in the database
+    """
+
+    def __init__(self, id, name, description, default_enabled=False, serializer_cls=None):
+        self.id = id
+        self.name = name
+        self.description = description
+        self.default_enabled = default_enabled
+        if serializer_cls is None:
+            self.serializer_cls = _FilterSerializer
+        else:
+            self.serializer_cls = serializer_cls
+
+
+def _get_filter_settings(relay_config, flt):
+    """
+    Gets the filter options from the relay config or the default option if not specified in the relay config
+
+    :param relay_config: the relay config for the request
+    :param flt: the filter
+    :return: the options for the filter
+    """
+    filter_settings = relay_config.config.get('filter_settings', {})
+    filter_key = flt.spec.id
+    return filter_settings.get(filter_key, None)
+
+
+def _is_filter_enabled(relay_config, flt):
+    filter_options = _get_filter_settings(relay_config, flt)
+
+    if filter_options is None:
+        return flt.spec.default_enabled
```

Comment:
Please try not to access `default_enabled` here, as this would mean that Relay has to replicate the defaults somehow.

Reply:
done